server/pulp/server/managers/auth/user/cud.py

```diff
@@ -111,35 +111,40 @@ def update_user(login, delta):
         # Check invalid values
         invalid_values = []
         if 'password' in delta:
-            if delta['password'] is None or invalid_type(delta['password'], basestring):
+            password = delta.pop('password')
+            if password is None or invalid_type(password, basestring):
                 invalid_values.append('password')
             else:
-                user['password'] = factory.password_manager().hash_password(delta['password'])
+                user['password'] = factory.password_manager().hash_password(password)
 
         if 'name' in delta:
-            if delta['name'] is None or invalid_type(delta['name'], basestring):
+            name = delta.pop('name')
+            if name is None or invalid_type(name, basestring):
                 invalid_values.append('name')
             else:
-                user['name'] = delta['name']
+                user['name'] = name
 
         if 'roles' in delta:
-            if delta['roles'] is None or invalid_type(delta['roles'], list):
+            roles = delta.pop('roles')
+            if roles is None or invalid_type(roles, list):
                 invalid_values.append('roles')
             else:
                 # Add new roles to the user and remove deleted roles from the user according to
                 # delta
                 role_manager = factory.role_manager()
                 old_roles = user['roles']
-                for new_role in delta['roles']:
+                for new_role in roles:
                     if new_role not in old_roles:
                         role_manager.add_user_to_role(new_role, login)
                 for old_role in old_roles:
-                    if old_role not in delta['roles']:
+                    if old_role not in roles:
                         role_manager.remove_user_from_role(old_role, login)
-                user['roles'] = delta['roles']
+                user['roles'] = roles
 
         if invalid_values:
             raise InvalidValue(invalid_values)
+        if delta:
+            raise InvalidValue(delta.keys())
```

Comment:
we have in ours docs:
400 - if one or more of the parameters is invalid
in my understanding when i mistype a value(or write in the data something that is not specified in the docs, like some random field) i would be prompted about that.
example: "error_message": "Invalid properties: ['pasword']"

Webpy is very inconsistent in this functionality, but most of the code is checking the params.
For example here:  https://github.com/pulp/pulp/blob/master/server/pulp/server/webservices/controllers/consumer_groups.py#L51
or here:
https://github.com/pulp/pulp/blob/master/server/pulp/server/managers/consumer/group/cud.py#L99
If I will remove that line and accidentally have some typo in the data, I  will have something like:

`$ curl -i -H "Accept: application/json" -H "WebFrameworkSwitch: django" -X PUT -k -u admin:admin -d '{"delta":{"pasword":"xxx"}}' "https://localhost/pulp/api/v2/users/user3/"
HTTP/1.1 200 OK
Date: Thu, 19 Feb 2015 10:13:21 GMT
Server: Apache/2.4.10 (Fedora) OpenSSL/1.0.1e-fips mod_wsgi/3.5 Python/2.7.5
Content-Length: 109
Content-Type: application/json

{"_id": {"$oid": "54e4686245ef48558f24fd70"}, "name": "user3", "roles": [], "_ns": "users", "login": "user3"}[ipanova@ina pulp]$`

The user will be prompted OK and he will even not notice that he has a typo in the 'password' and actually it was not updated.
In addition to this, that line fixes the bug  https://bugzilla.redhat.com/show_bug.cgi?id=1193059